https.go

```diff
@@ -667,6 +672,11 @@ func (proxy *ProxyHttpServer) HandleHttps(w http.ResponseWriter, r *http.Request
 				ctx.Req = req
 
 				req, resp := proxy.filterRequest(req, ctx)
+				// If a cancel function is set, ensure we call it when
```

Comment:
It's not entirely clear to me why cancel needs to be called here. I might be thrown off by the name. I wonder if ctx.Cancel() would better describe itself with a different name. It seems like it's used as a callback indicating request handling has completed. I guess we just happen to want to cancel a context when that happens? To make it make better sense generally, what about a name like "ctx.HandlerDone()" or "ctx.HandlerDoneCallback()" ?

Reply:
This is called here under the circumstances where a `context.Context()` is defined as part of a callback function defined in `ProxyConds`; the reason it's called here is because `proxy.filterRequest()` is what triggers the callback function to be called.

Since this context should be cancelled at the end of the request, we cannot do this in the callback function defined by the consumer:

```go
ctx, cancel := context.WithCancel(ctx)
defer cancel()
```

And must call `defer cancel()` immediately after it is initialized in `goproxy`.

Reply:
For clarity, we could call it `ProxyCtx.ProxyCondsCancel` or something instead.

Reply:
How about `ProxyCtx.FilterRequestDone()`? I might still be really missing something but there's nothing about this as I understand it so far that makes it necessarily a cancel function, is there? That's the caller's decision. I'm motivated to be as agnostic as possible as far as what the caller is doing with it.

Reply:
Or "Done" or "RequestDone" something more descriptive of what's happening rather than what you're supposed to be doing about it. Of course there is the convention of `ctx.Done()` being returning a channel which could cause confusion.

Reply:
@jessemalone I think that's more confusing, since this is a CancelFunc, and we're making sure we call `cancel()` on a context we've defined

Reply:
Fair enough. If an additional use for it comes it it can always be revisited.